src/app/ApplicationTemplate.Tests.Functional/FunctionalTestBase.cs

```diff
@@ -182,4 +194,21 @@ async Task IAsyncLifetime.DisposeAsync()
 	{
 		_coreStartup.Dispose();
 	}
+
+	private sealed class BackButtonSource : IBackButtonSource
+	{
+		public string Name => "Test";
```

Comment:
You can probably give a better name and also adjust the class name too.
this implementation is specific for functional tests so that should help to find a good name.